azimuth/plots/dataset_warnings.py

```diff
@@ -432,99 +434,83 @@ def class_representation(
 
 
 def create_histogram_mean_std(
-    train_count: np.ndarray,
-    eval_count: np.ndarray,
-    threshold_mean: float,
-    threshold_std: float,
-    train_stats: Tuple[np.ndarray, np.ndarray],
-    eval_stats: Tuple[np.ndarray, np.ndarray],
-    divergence_mean: Optional[float],
-    divergence_std: Optional[float],
-    label_name: Optional[str] = None,
+    hist_per_split: Dict[DatasetSplitName, np.ndarray],
+    value_per_agg_per_split: Dict[DatasetSplitName, Dict[Agg, float]],
+    divergence_per_agg: Optional[Dict[Agg, float]] = None,
 ) -> PlotSpecification:
     """Create the histogram traces and annotations for each label (and "all").
 
     Args:
-        train_count: Histogram values for the selected label in the train set.
-        eval_count: Histogram values for the selected label in the evaluation set.
-        threshold_mean: Alert value.
-        threshold_std: Alert value.
-        train_stats: Mean and std dev of the train.
-        eval_stats: Mean and std dev of the evaluation set.
-        divergence_mean: Difference in the mean. None means one distribution is empty.
-        divergence_std: Difference in the std dev. None means one distribution is empty.
-        label_name: Name of the label, when sending 'None', the traces are generated for 'all'.
+        hist_per_split: Histogram values for the selected label per split.
+        value_per_agg_per_split: Mean and std dev for the selected label per split.
+        divergence_per_agg: Difference in the mean and std dev. None means that we are generating
+            the plot for 'all' labels.
 
     Returns:
         Plot for one label.
 
     """
     fig = go.Figure()
 
-    train_count_norm = train_count / max(sum(train_count), 1)
-    eval_count_norm = eval_count / max(sum(eval_count), 1)
+    hist_normalized = {split: hist / max(sum(hist), 1) for split, hist in hist_per_split.items()}
```

Comment:
I have an idea! We can filter out the empty histograms from `hist_normalized`. That also solves the `max(..., 1)` hack to avoid divisions by 0.
```suggestion
    hist_normalized = {split: hist / sum(hist) for split, hist in hist_per_split.items() if any(hist)}
```
Then, there is no need to for this `if` inside the loop anymore:
```diff
-        if hist.max() != 0:
```
And the two checks become easy:
```diff
-    two_distributions = all(any(value > 0 for value in h) for h in hist_normalized.values())
- 
-    if divergence_per_agg and two_distributions:
+    if divergence_per_agg and len(hist_normalized) == 2:
```
```diff
-    empty_distribution = max(value.max() for value in hist_normalized.values()) == 0
-    if empty_distribution:
+    if len(hist_normalized) == 0:
```
I think there is only one more thing to adjust for this to work; avoiding the error in `max()` if there are no non-empty histograms:
```diff
     # Helps position the error bars.
-    max_y = max(value.max() for value in hist_normalized.values())
+    max_y = max((value.max() for value in hist_normalized.values()), default=0)
```

Reply:
Good idea